woocommerce/woocommerce-android
# Issue/11899 new image selector

Closes: #11899 


### Description

Add all the logic to handle adding, replacing or removing and product image to be scanned and extract a prompt from it. 
Not that this PR only adds the logic to manage the image and update the UI accordingly, the scanning of the image will be tackled in a different PR. 

### Testing information

1. Trigger product creation with AI flow
2. Add an image by clicking on `Read text from product photo`
3. Check that the UI is updated with the image
4. Click on the three options available to manage the image: `view`, `replace`, `remove`. 

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/2663464/e8897eb2-43e3-4dcf-b360-8e2188599723
